manifests/params.pp

```diff
@@ -2,10 +2,14 @@
 # @api private
 class dhcp::params {
 
-  $dnsdomain = [$facts['domain']]
+  $dnsdomain = $facts['os']['family'] ? {
+    'Archlinux' => [$facts['domain']], # lint:ignore:legacy_facts
```

Comment:
@ekohl any tips on debugging why the modern domain fact is not working for Archlinux? Do you suspect that it is an issue in https://github.com/mcanevet/rspec-puppet-facts or is there a better place to start?

Reply:
```
[wclark@prometheus facterdb]$ sudo facterdb 'facterversion=/^2.4\./ and operatingsystem=Archlinux' | grep domain
[DEPRECATION] `get_os_facts` is deprecated. Please use `get_facts(facterversion=/^2.4\./ and operatingsystem=Archlinux)` instead.
    "domain": "example.com",
    "domain": "example.com"
```

(If you inspect the file you can see that is truly the legacy domain fact, it's not nested under networking like the modern fact)

Checking the table at https://github.com/camptocamp/facterdb/blob/master/README.md it seems that more recent `facter` versions are not currently supported for Archlinux

Reply:
We should be setting the facter version. The released version of rspec-puppet-facts uses the facts of the installed gem version. We should be setting the facter version to match the released versions. master has code for it but it's unreleased. https://github.com/voxpupuli/voxpupuli-test has helpers for that. I'd like to migrate to it instead of shipping it in our spec helper.